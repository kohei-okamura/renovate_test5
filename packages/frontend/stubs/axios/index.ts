/*
 * Copyright © 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { NuxtAxiosInstance } from '@nuxtjs/axios'
import { stubAttendances } from '~~/stubs/axios/stub-attendances'
import { stubCallings } from '~~/stubs/axios/stub-callings'
import { stubCopayLists } from '~~/stubs/axios/stub-copay-lists'
import { stubDwsAreaGrades } from '~~/stubs/axios/stub-dws-area-grades'
import { stubDwsBillingCopayCoordinations } from '~~/stubs/axios/stub-dws-billing-copay-coordination'
import { stubDwsDwsBillingServiceReports } from '~~/stubs/axios/stub-dws-billing-service-reports'
import { stubDwsBillings } from '~~/stubs/axios/stub-dws-billings'
import { stubDwsCertifications } from '~~/stubs/axios/stub-dws-certifications'
import { stubDwsContracts } from '~~/stubs/axios/stub-dws-contracts'
import { stubDwsProjectServiceMenus } from '~~/stubs/axios/stub-dws-project-service-menus'
import { stubDwsProjects } from '~~/stubs/axios/stub-dws-projects'
import { stubDwsProvisionReports } from '~~/stubs/axios/stub-dws-provision-reports'
import { stubDwsSubsidies } from '~~/stubs/axios/stub-dws-subsidies'
import { stubHomeHelpServiceCalcSpecs } from '~~/stubs/axios/stub-home-help-service-calc-specs'
import { stubHomeVisitLongTermCareCalcSpecs } from '~~/stubs/axios/stub-home-visit-long-term-care-calc-specs'
import { stubInvitations } from '~~/stubs/axios/stub-invitations'
import { stubLtcsAreaGrades } from '~~/stubs/axios/stub-ltcs-area-grades'
import { stubLtcsBillings } from '~~/stubs/axios/stub-ltcs-billings'
import { stubLtcsContracts } from '~~/stubs/axios/stub-ltcs-contracts'
import { stubLtcsHomeVisitLongTermCareDictionary } from '~~/stubs/axios/stub-ltcs-home-visit-long-term-care-dictionary'
import { stubLtcsInsCards } from '~~/stubs/axios/stub-ltcs-ins-cards'
import { stubLtcsProjectServiceMenus } from '~~/stubs/axios/stub-ltcs-project-service-menus'
import { stubLtcsProjects } from '~~/stubs/axios/stub-ltcs-projects'
import { stubLtcsProvisionReports } from '~~/stubs/axios/stub-ltcs-provision-reports'
import { stubLtcsSubsidies } from '~~/stubs/axios/stub-ltcs-subsidies'
import { stubOfficeGroups } from '~~/stubs/axios/stub-office-groups'
import { stubOffices } from '~~/stubs/axios/stub-offices'
import { stubOptions } from '~~/stubs/axios/stub-options'
import { stubOwnExpensePrograms } from '~~/stubs/axios/stub-own-expense-programs'
import { stubPasswordResets } from '~~/stubs/axios/stub-password-resets'
import { stubPermissions } from '~~/stubs/axios/stub-permissions'
import { stubPostcodes } from '~~/stubs/axios/stub-postcodes'
import { stubRoles } from '~~/stubs/axios/stub-roles'
import { stubSessions } from '~~/stubs/axios/stub-sessions'
import { stubSetting } from '~~/stubs/axios/stub-setting'
import { stubShifts } from '~~/stubs/axios/stub-shifts'
import { stubStaffVerifications } from '~~/stubs/axios/stub-staff-verifications'
import { stubStaffs } from '~~/stubs/axios/stub-staffs'
import { stubUserBillings } from '~~/stubs/axios/stub-user-billings'
import { stubUserDwsCalcSpecs } from '~~/stubs/axios/stub-user-dws-calc-specs'
import { stubUserLtcsCalcSpecs } from '~~/stubs/axios/stub-user-ltcs-calc-specs'
import { stubUsers } from '~~/stubs/axios/stub-users'
import { stubVisitingCareForPwsdCalcSpecs } from '~~/stubs/axios/stub-visiting-care-for-pwsd-calc-specs'
import { stubWithdrawalTransactions } from '~~/stubs/axios/stub-withdrawal-transactions'
import { createMockAdapter } from '~~/stubs/axios/utils'

/**
 * Axios からの API リクエストをスタブ化する.
 */
export const stubAxios = ($axios: NuxtAxiosInstance): void => {
  const mockAdapter = createMockAdapter($axios)
  const stubFunctions = [
    stubAttendances,
    stubCallings,
    stubCopayLists,
    stubDwsAreaGrades,
    stubDwsBillingCopayCoordinations,
    stubDwsBillings,
    stubDwsCertifications,
    stubDwsContracts,
    stubDwsDwsBillingServiceReports,
    stubDwsProjects,
    stubDwsProjectServiceMenus,
    stubDwsProvisionReports,
    stubDwsSubsidies,
    stubHomeHelpServiceCalcSpecs,
    stubHomeVisitLongTermCareCalcSpecs,
    stubInvitations,
    stubLtcsAreaGrades,
    stubLtcsBillings,
    stubLtcsContracts,
    stubLtcsHomeVisitLongTermCareDictionary,
    stubLtcsInsCards,
    stubLtcsProjects,
    stubLtcsProjectServiceMenus,
    stubLtcsProvisionReports,
    stubLtcsSubsidies,
    stubOfficeGroups,
    stubOffices,
    stubOptions,
    stubOwnExpensePrograms,
    stubPasswordResets,
    stubPermissions,
    stubPostcodes,
    stubRoles,
    stubSessions,
    stubSetting,
    stubShifts,
    stubStaffs,
    stubStaffVerifications,
    stubUserBillings,
    stubUserDwsCalcSpecs,
    stubUserLtcsCalcSpecs,
    stubUsers,
    stubVisitingCareForPwsdCalcSpecs,
    stubWithdrawalTransactions
  ]
  stubFunctions.forEach(f => f(mockAdapter))
}
