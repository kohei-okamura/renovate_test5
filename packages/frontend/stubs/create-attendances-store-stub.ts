/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { reactive } from '@nuxtjs/composition-api'
import * as m from '~/composables/stores/use-attendances-store'
import { createStore } from '~/composables/stores/utils'

type Data = m.AttendancesData
type Store = m.AttendancesStore

export const createAttendancesStoreStub = (data: Partial<Data> = {}): Store => {
  const state = reactive({
    ...m.createAttendancesState(),
    ...data
  })
  const actions = {
    getIndex: () => Promise.resolve()
  }
  const store: Store = createStore({ actions, state })
  jest.spyOn(m, 'useAttendancesStore').mockReturnValue(store)
  jest.spyOn(store, 'getIndex').mockResolvedValue()
  return store
}
