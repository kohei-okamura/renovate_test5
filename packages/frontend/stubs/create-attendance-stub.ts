/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Attendance, AttendanceId } from '~/models/attendance'
import { Contract } from '~/models/contract'
import { Shift } from '~/models/shift'
import {
  createShiftStub,
  createShiftStubs,
  createShiftStubsForContract,
  SHIFT_ID_MAX,
  SHIFT_ID_MIN,
  SHIFT_STUB_COUNT
} from '~~/stubs/create-shift-stub'
import { CreateStubs } from '~~/stubs/index'

export const ATTENDANCE_ID_MAX = SHIFT_ID_MAX
export const ATTENDANCE_ID_MIN = SHIFT_ID_MIN
export const ATTENDANCE_STUB_COUNT = SHIFT_STUB_COUNT

export const createAttendanceStubFromShift = (shift: Shift): Attendance => ({
  ...shift,
  shiftId: shift.id
})

export function createAttendanceStub (id: AttendanceId, contract: Contract): Attendance {
  return createAttendanceStubFromShift(createShiftStub(id, contract))
}

export function createAttendanceStubsForContract (contract: Contract): Attendance[] {
  return createShiftStubsForContract(contract).map(createAttendanceStubFromShift)
}

export const createAttendanceStubs: CreateStubs<Attendance> = (n = ATTENDANCE_STUB_COUNT, skip = 0) => {
  return createShiftStubs(n, skip).map(createAttendanceStubFromShift)
}
