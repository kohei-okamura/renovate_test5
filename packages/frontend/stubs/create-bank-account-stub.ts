/*
 * Copyright Â© 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { BankAccountType } from '@zinger/enums/lib/bank-account-type'
import { BankAccount, BankAccountId } from '~/models/bank-account'
import { ID_MAX, ID_MIN, SEEDS } from '~~/stubs'
import { createFaker } from '~~/stubs/fake'
import { banks, blanches } from '~~/stubs/fake/banks'

export const BANK_ID_MAX = ID_MAX
export const BANK_ID_MIN = ID_MIN

export function createBankAccountStub (id: BankAccountId = BANK_ID_MIN): BankAccount {
  const faker = createFaker(SEEDS[id - 1])
  const fake = faker.createFake()
  const bank = faker.randomElement(banks)
  const blanch = faker.randomElement(blanches[bank.code])
  return {
    id,
    bankName: bank.name,
    bankCode: bank.code,
    bankBranchName: blanch.name,
    bankBranchCode: blanch.code,
    bankAccountType: faker.randomElement([
      BankAccountType.ordinaryDeposit,
      BankAccountType.currentDeposit,
      BankAccountType.fixedDeposit
    ]),
    bankAccountNumber: faker.randomNumericString(7),
    bankAccountHolder: fake.name.phoneticDisplayName,
    createdAt: faker.randomDateTimeString(),
    updatedAt: faker.randomDateTimeString()
  }
}

export function createEmptyBankAccountStub (id: BankAccountId = BANK_ID_MIN): BankAccount {
  return {
    id,
    bankName: '',
    bankCode: '',
    bankBranchName: '',
    bankBranchCode: '',
    bankAccountType: BankAccountType.ordinaryDeposit,
    bankAccountNumber: '',
    bankAccountHolder: '',
    createdAt: '',
    updatedAt: ''
  }
}
