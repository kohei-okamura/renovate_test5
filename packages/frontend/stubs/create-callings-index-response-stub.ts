/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Shift } from '~/models/shift'
import { Api } from '~/services/api/core'
import { ShiftsApi } from '~/services/api/shifts-api'
import { createCallingsStub } from '~~/stubs/create-callings-stub'
import { createIndexResponse } from '~~/stubs/create-index-response'
import { SHIFT_STUB_COUNT } from '~~/stubs/create-shift-stub'

export function createCallingIndexResponseStub (params: ShiftsApi.GetIndexParams = {}): Api.GetIndexResponse<Shift> {
  return createIndexResponse(params, SHIFT_STUB_COUNT, createCallingsStub)
}
