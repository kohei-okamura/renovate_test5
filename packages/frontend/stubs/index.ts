/*
 * Copyright Â© 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { createFaker } from '~~/stubs/fake'

export const ID_MAX = 100
export const ID_MIN = 1
export const STUB_COUNT = 1000
export const STUB_DEFAULT_SEED = 'zinger'

export const createSeeds = (n: number, seed: string = STUB_DEFAULT_SEED): string[] => {
  const faker = createFaker(seed)
  return Array.from(Array(n), () => faker.randomString(8))
}

export const SEEDS = createSeeds(STUB_COUNT * 3)

export type CreateStubs<T> = {
  (n?: number, skip?: number): T[]
}
