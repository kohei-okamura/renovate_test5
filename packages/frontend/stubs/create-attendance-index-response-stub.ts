/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { AttendancesApi } from '~/services/api/attendances-api'
import { ATTENDANCE_STUB_COUNT, createAttendanceStubs } from '~~/stubs/create-attendance-stub'
import { createIndexResponse } from '~~/stubs/create-index-response'

export function createAttendanceIndexResponseStub (
  params: AttendancesApi.GetIndexParams = {}
): AttendancesApi.GetIndexResponse {
  return createIndexResponse(params, ATTENDANCE_STUB_COUNT, createAttendanceStubs)
}
