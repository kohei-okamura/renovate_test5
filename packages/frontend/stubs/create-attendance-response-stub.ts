/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { AttendanceId } from '~/models/attendance'
import { AttendancesApi } from '~/services/api/attendances-api'
import { ATTENDANCE_ID_MIN, createAttendanceStub } from '~~/stubs/create-attendance-stub'
import { createContractStub } from '~~/stubs/create-contract-stub'

export function createAttendanceResponseStub (id: AttendanceId = ATTENDANCE_ID_MIN): AttendancesApi.GetResponse {
  const contractId = Math.floor(id / 10)
  const contract = createContractStub(contractId)
  const attendance = createAttendanceStub(id, contract)
  return {
    attendance
  }
}
