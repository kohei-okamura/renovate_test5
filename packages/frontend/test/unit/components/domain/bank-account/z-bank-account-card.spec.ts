/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { MountOptions, Wrapper } from '@vue/test-utils'
import Vue from 'vue'
import ZBankAccountCard from '~/components/domain/bank-account/z-bank-account-card.vue'
import { BANK_ID_MIN, createBankAccountStub } from '~~/stubs/create-bank-account-stub'
import { setupComponentTest } from '~~/test/helpers/setup-component-test'

describe('z-bank-account-card.vue', () => {
  const { mount } = setupComponentTest()
  let wrapper: Wrapper<Vue>

  function mountComponent (options: MountOptions<Vue>) {
    wrapper = mount(ZBankAccountCard, options)
  }

  it('should be rendered correctly without title', () => {
    const propsData = {
      ...createBankAccountStub(BANK_ID_MIN)
    }
    mountComponent({ propsData })
    expect(wrapper).toMatchSnapshot()
  })

  it('should be rendered correctly with title', () => {
    const propsData = {
      ...createBankAccountStub(BANK_ID_MIN + 1),
      title: '給与振込口座'
    }
    mountComponent({ propsData })
    expect(wrapper).toMatchSnapshot()
  })
})
