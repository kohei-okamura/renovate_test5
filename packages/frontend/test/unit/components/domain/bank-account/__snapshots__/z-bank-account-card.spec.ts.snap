// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`z-bank-account-card.vue should be rendered correctly with title 1`] = `
<div
  class="z-data-card-stub"
  createdat="1973-07-07T03:22:07+0900"
  id="2"
  title="給与振込口座"
  updatedat="2024-05-24T11:53:42+0900"
>
  <div
    class="z-data-card-stub-slot-actions"
  />
  <z-data-card-item-stub
    icon="M6.5,10H4.5V17H6.5V10M12.5,10H10.5V17H12.5V10M21,19H2V21H21V19M18.5,10H16.5V17H18.5V10M11.5,3.26L16.71,6H6.29L11.5,3.26M11.5,1L2,6V8H21V6L11.5,1Z"
    label="銀行名（銀行コード）"
  >
    埼玉りそな（0017）
  </z-data-card-item-stub>
  <z-data-card-item-stub
    label="支店名（支店コード）"
  >
    土呂出張所（337）
  </z-data-card-item-stub>
  <z-data-card-item-stub
    label="銀行口座種別"
    value="定期預金"
  />
  <z-data-card-item-stub
    label="口座番号"
    value="1657959"
  />
  <z-data-card-item-stub
    label="口座名義"
    value="キダ サトミ"
  />
  <!---->
  <div
    class="z-data-card-stub-slot-header"
  />
</div>
`;

exports[`z-bank-account-card.vue should be rendered correctly without title 1`] = `
<div
  class="z-data-card-stub"
  createdat="1990-08-08T01:31:13+0900"
  id="1"
  title="銀行口座"
  updatedat="2014-07-01T18:07:18+0900"
>
  <div
    class="z-data-card-stub-slot-actions"
  />
  <z-data-card-item-stub
    icon="M6.5,10H4.5V17H6.5V10M12.5,10H10.5V17H12.5V10M21,19H2V21H21V19M18.5,10H16.5V17H18.5V10M11.5,3.26L16.71,6H6.29L11.5,3.26M11.5,1L2,6V8H21V6L11.5,1Z"
    label="銀行名（銀行コード）"
  >
    ソニー（0035）
  </z-data-card-item-stub>
  <z-data-card-item-stub
    label="支店名（支店コード）"
  >
    本店（001）
  </z-data-card-item-stub>
  <z-data-card-item-stub
    label="銀行口座種別"
    value="当座預金"
  />
  <z-data-card-item-stub
    label="口座番号"
    value="6614772"
  />
  <z-data-card-item-stub
    label="口座名義"
    value="アライ エリカ"
  />
  <!---->
  <div
    class="z-data-card-stub-slot-header"
  />
</div>
`;
